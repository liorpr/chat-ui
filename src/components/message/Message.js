
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Avatar from '../avatar/Avatar';

export default class Message extends PureComponent {  
  
  static propTypes = {
    username: PropTypes.string,
    message: PropTypes.string,
    avatar: PropTypes.string,
    isSelfMessage: PropTypes.bool
  };
  
  render() {
    const { username, message, avatar, isSelfMessage: self } = this.props;
    return (
      <li className={classNames('message-container', { self })}>
        <Avatar name={avatar} />
        <div className="content">
          <p className={classNames('box', { self })}>
            { message }
          </p>
          <div className={classNames('username', { self })}>
            { username }
          </div>
        </div>
      </li>
    );
  }
}