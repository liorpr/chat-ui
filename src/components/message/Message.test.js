
import React from 'react';
import TestRenderer from 'react-test-renderer';
import Message from './message';

const username = 'username';
const message = 'message';
const avatar = 'avatar';

it('renders correctly', () => {
  const tree = TestRenderer
    .create(<Message
        username={username}
        message={message}
        avatar={avatar}
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});