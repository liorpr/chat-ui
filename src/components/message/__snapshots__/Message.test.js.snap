// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="message-container"
>
  <img
    alt="avatar"
    className="avatar-img"
    src="https://spotim-demo-chat-server.herokuapp.com/avatars/avatar.png"
  />
  <div
    className="content"
  >
    <p
      className="box"
    >
      message
    </p>
    <div
      className="username"
    >
      username
    </div>
  </div>
</li>
`;
